bftengine/src/bftengine/KeyExchangeManager.cpp

```diff
@@ -113,6 +116,42 @@ void KeyExchangeManager::loadPublicKeys() {
   notifyRegistry();
 }
 
+void KeyExchangeManager::exchangeTlsKeys(const SeqNum& bft_sn) {
+  auto repId = bftEngine::ReplicaConfig::instance().replicaId;
+  auto keys = concord::util::crypto::Crypto::instance().generateECDSAKeyPair(
+      concord::util::crypto::KeyFormat::PemFormat, concord::util::crypto::CurveType::secp384r1);
+  auto cert = concord::util::crypto::Crypto::instance().generateSelfSignedCertificate(
+      keys, "node" + std::to_string(repId) + "ser", repId);
+  std::string pk_path =
+      bftEngine::ReplicaConfig::instance().certificatesRootPath + "/" + std::to_string(repId) + "/server/pk.pem";
+  concord::secretsmanager::SecretsManagerPlain psm_;
+  std::fstream nec_f(pk_path);
+  if (nec_f.good()) {
+    psm_.encryptFile(pk_path, keys.first);
+  }
+  std::fstream ec_f(pk_path + ".enc");
+  if (ec_f.good()) {
+    secretsMgr_->encryptFile(pk_path + ".enc", keys.first);
+  }
+
+  concord::messages::ReconfigurationRequest req;
```

Comment:
Why to encapsulate `ReplicaTlsExchangeKey` in a `ReconfigurationRequest`?
`ReconfigurationRequest` is usually sent by an operator.

Reply:
As a response to the operator `reconfiguraitonRequest`, the exchanged replicas send another `reconfigurationRequest` that contains the new certificate.
This is to avoid having a different execution handler for this method (like we have with regular key exchange) and to work with a single framework